spark/dl/src/main/scala/com/intel/analytics/bigdl/optim/DistriOptimizer.scala

```diff
@@ -71,7 +72,8 @@ object DistriOptimizer {
     localStates: Array[Table],
     var moduleTimeList: Array[Long] = null,
     localMethods: Array[Option[Array[ValidationMethod[T]]]],
-    optimMethods: Map[String, OptimMethod[T]]
+    optimMethods: Map[String, OptimMethod[T]],
+    parameterSynchronizer: DistriParameterSynchronizer[T] = null
```

Comment:
why we cache this?

Reply:
We want to release all the resources at last and also we need to reference this object to get initially set partition ID